src/test/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApprovalTest.java

```diff
@@ -176,6 +176,29 @@ public void upgradeSmokes() throws Exception {
                 r.assertBuildStatus(Result.SUCCESS, p.scheduleBuild2(0).get()));
     }
 
+    @LocalData // Some scriptApproval.xml with existing signatures approved
+    @Test
+    public void reload() throws Exception {
+        configureSecurity();
+        ScriptApproval sa = ScriptApproval.get();
+
+        FreeStyleProject p = r.createFreeStyleProject();
+        p.getPublishersList().add(new TestGroovyRecorder(
+                new SecureGroovyScript("jenkins.model.Jenkins.instance", true, null)));
+        p.getPublishersList().add(new TestGroovyRecorder(
+                new SecureGroovyScript("println(jenkins.model.Jenkins.instance.getLabels())", false, null)));
+        r.assertLogNotContains("org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException: "
+                        + "Scripts not permitted to use staticMethod jenkins.model.Jenkins getInstance",
+                r.assertBuildStatus(Result.SUCCESS, p.scheduleBuild2(0).get()));
```

Comment:
Perhaps besides the point since you approved the whole script anyway, and if you had not, you would also get an error on `method jenkins.model.Jenkins getLabels`, but OK.